src/sentry/static/sentry/app/views/performance/data.tsx

```diff
@@ -47,6 +49,15 @@ export function generatePerformanceQuery(location: Location): Readonly<NewQuery>
         : '-rpm';
   }
 
+  if (query?.query) {
+    const searchQuery = decodeScalar(query.query);
+    if (searchQuery) {
+      extra.query = `${PERFORMANCE_EVENT_VIEW.query} transaction:"*${escapeDoubleQuotes(
+        searchQuery
+      )}*"`;
```

Comment:
Should we be using the querystring module for this? We could tokenize the existing query, add in the user's text and then serialize the string again?

Reply:
Yea we should.

Reply:
Should be addressed now.